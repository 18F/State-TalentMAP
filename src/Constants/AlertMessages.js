export const NO_LANGUAGES = 'None listed';
export const NO_POST = 'None listed';
export const NO_POST_DIFFERENTIAL = 'None listed';
export const NO_DANGER_PAY = 'None listed';
export const NO_COLA = 'None listed';
export const NO_TOUR_OF_DUTY = 'None listed';
