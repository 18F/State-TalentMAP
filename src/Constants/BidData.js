// prop names from the API
export const APPROVED_PROP = 'approved';
export const CLOSED_PROP = 'closed';
export const DRAFT_PROP = 'draft';
export const DECLINED_PROP = 'declined';
export const HAND_SHAKE_ACCEPTED_PROP = 'handshake_accepted';
export const HAND_SHAKE_OFFERED_PROP = 'handshake_offered';
export const HAND_SHAKE_DECLINED_PROP = 'handshake_declined';
export const PRE_PANEL_PROP = 'pre_panel';
export const IN_PANEL_PROP = 'in_panel';
export const SUBMITTED_PROP = 'submitted';

export const DRAFT_NUMBER = 1;
export const SUBMITTED_NUMBER = 2;
export const HAND_SHAKE_OFFERED_NUMBER = 3;
export const HAND_SHAKE_ACCEPTED_NUMBER = 4;
export const IN_PANEL_NUMBER = 5;
export const APPROVED_NUMBER = 6;

// We make all of them functions so that we don't have to guess whethe to call a function or not
export const DRAFT_TITLE = () => ['Draft'];
export const SUBMIT_BID_ACTION_TITLE = () => ['Submit Bid'];
export const SUBMIT_BID_COMPLETE_TITLE = (bid = {}) => ['Bid Submitted', bid.submission_date];
export const HAND_SHAKE_EVALUATE_TITLE = () => ['Evaluate Bid'];
export const HAND_SHAKE_EVALUATING_TITLE = () => ['Bid being evaluated'];
export const HAND_SHAKE_OFFERED_TITLE = (bid = {}) => ['Handshake Offered', bid.hand_shake_offered_date];
export const HAND_SHAKE_ACCEPTED_TITLE = (bid = {}) => ['Handshake Accepted', bid.hand_shake_accepted_date];
export const PANEL_TITLE = (bid = {}) => ['Panel',
  `${bid.scheduled_panel_date ? `Scheduled: ${bid.scheduled_panel_date}` : ''}`,
  `${bid.panel_date ? `Panel date: ${bid.panel_date}` : ''}`];
export const APPROVAL_TITLE = () => ['Approval'];
