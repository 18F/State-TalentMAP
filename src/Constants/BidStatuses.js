// store any bid statuses from the API here
// class_name is used to pass to parent container for conditional css class name
export const APPROVED = { property: 'approved', class_name: 'approved', text: 'Approved' };
export const CLOSED = { property: 'closed', class_name: 'closed', text: 'Closed' };
export const DRAFT = { property: 'draft', class_name: 'draft', text: 'Draft Bid' };
export const DECLINED = { property: 'declined', class_name: 'declined', text: 'Bid Unapproved' };
export const HAND_SHAKE_ACCEPTED = { property: 'handshake_accepted', class_name: 'hand-shake-accepted', text: 'Handshake Accepted' };
export const HAND_SHAKE_OFFERED = { property: 'handshake_offered', class_name: 'hand-shake-offered', text: 'Handshake Offered' };
export const IN_PANEL = { property: 'in_panel', class_name: 'in-panel', text: 'In Panel' };
export const SUBMITTED = { property: 'submitted', class_name: 'submitted', text: 'Bid Submitted' };

const statusArray = [
  APPROVED, CLOSED, DRAFT, DECLINED, HAND_SHAKE_ACCEPTED, HAND_SHAKE_OFFERED, IN_PANEL, SUBMITTED,
];

// find the correct object based on status, then return the class_name
export const getStatusClass = (statusName) => {
  const matchingStatus = statusArray.find(s => s.property === statusName);
  return matchingStatus ? matchingStatus.class_name : '';
};
