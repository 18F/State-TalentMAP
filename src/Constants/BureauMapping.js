export const BUREAU_MAPPING = [
  /* {
    id: 1,
    description: 'Regional Bureaus',
    codeGroup: ['110000', '120000', '130000', '140000', '146000', '150000', '160000'],
    isRegional: true,
  }, */
  {
    id: '2',
    description: 'Arms Control and International Security',
    codeGroup: ['014540', '010164', '014000'],
  },
  {
    id: '3',
    description: 'Civilian Security, Democracy, and Human Rights',
    codeGroup: ['013000', '033510', '018000', '031000', '010113', '011810'],
  },
  {
    id: '4',
    description: 'Democratic Republic of Congo',
    codeGroup: ['010108', '010183', '010106', '010145'],
  },
  {
    id: '5',
    description: 'Economic Growth, Energy, and the Environment',
    codeGroup: ['040700', '041600', '011701', '021250'],
  },
  {
    id: '6',
    description: 'Management',
    codeGroup: ['201000', '041900', '170100', '010623', '016000', '241000', '012900', '012200'],
  },
  {
    id: '7',
    description: 'Public Diplomacy and Public Affairs',
    codeGroup: ['100100'],
  },
  {
    id: '8',
    description: 'Special Representatives, Envoys, Advisors & Coordinators',
    codeGroup: ['011210', '011702', '010173', '010114', '010135', '010170', '010163', '010140', '010182', '010179', '010160', '010181'],
  },
];

export const getBureauCodeGroup = () => {
  let codeGroup = [];
  BUREAU_MAPPING.forEach((group) => {
    codeGroup = codeGroup.concat(group.codeGroup);
  });
  return codeGroup.toString();
};
