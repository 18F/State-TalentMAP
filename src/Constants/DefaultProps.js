export const ACCORDION_SELECTION = { main: '', sub: '' };

export const DEFAULT_HOME_PAGE_POSITIONS = {
  isHighlighted: [],
  isNew: [],
};

export const DEFAULT_USER_PROFILE = {
  user: {
    username: '...', // show '...' when loading
    first_name: '',
    last_name: '',
  },
  favorite_positions: [],
};

export const POSITION_RESULTS_OBJECT = {
  results: [],
};

export default ACCORDION_SELECTION;
