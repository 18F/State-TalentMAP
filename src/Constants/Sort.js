const POSITION_SEARCH_SORTS = {
  options: [
    { value: '', text: 'Select a sort option', disabled: true },
    { value: 'create_date', text: 'Position Posted Date (New - Old)' },
    { value: 'post__has_service_needs_differential', text: 'Service Need Differential (T - F)' },
    { value: 'bureau', text: 'Bureau (A - Z)' },
  ],
};

POSITION_SEARCH_SORTS.defaultSort = POSITION_SEARCH_SORTS.options[0].value;

export default POSITION_SEARCH_SORTS;
