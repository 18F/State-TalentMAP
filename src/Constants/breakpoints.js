// should match responsive breakpoints in _variables.scss
const breakpoints = {
  screenSmMin: 600,
  get screenXsMax() { return this.screenSmMin - 1; },
  screenMdMin: 951,
  get screenSmMax() { return this.screenMdMin - 1; },
  screenLgMin: 1200,
  get screenMdMax() { return this.screenLgMin - 1; },
};

export default breakpoints;
