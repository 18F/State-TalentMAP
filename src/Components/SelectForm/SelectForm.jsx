import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { SORT_BY_ARRAY } from '../../Constants/PropTypes';

class SelectForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selection: { value: this.props.defaultSort || '' },
    };
  }

  selectOption(e) {
    const selection = e.target.value;
    this.setState({ selection });
    this.props.onSelectOption(e);
  }
  render() {
    const { id, label, options } = this.props;
    const optionList = options.map(option =>
      (
        <option
          key={option.value}
          disabled={option.disabled}
          value={option.value}
        >
          {option.text}
        </option>
      ),
    );
    return (
      <form className="usa-form">
        <label htmlFor={id}>{label}</label>
        <select
          name={id}
          id={id}
          onChange={e => this.selectOption(e)}
          value={this.state.selection.value}
        >
          {
            optionList
          }
        </select>
      </form>
    );
  }
}

SelectForm.propTypes = {
  id: PropTypes.node.isRequired,
  label: PropTypes.node.isRequired,
  defaultSort: PropTypes.node,
  options: SORT_BY_ARRAY.isRequired,
  onSelectOption: PropTypes.func.isRequired,
};

SelectForm.defaultProps = {
  languages: [],
  defaultSort: '',
};

export default SelectForm;
