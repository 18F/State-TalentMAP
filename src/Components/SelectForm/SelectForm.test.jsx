import { shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';
import toJSON from 'enzyme-to-json';
import SelectForm from './SelectForm';
import { POSITION_SEARCH_SORTS } from '../../Constants/Sort';

describe('SelectForm', () => {
  let wrapper = null;

  it('is defined', () => {
    wrapper = shallow(<SelectForm
      id={1}
      label="Some label"
      defaultSort=""
      options={POSITION_SEARCH_SORTS.options}
      onSelectOption={() => {}}
    />);
    expect(wrapper).toBeDefined();
  });

  it('can receive different props', () => {
    wrapper = shallow(<SelectForm
      id={2}
      label="Some label"
      defaultSort=""
      options={POSITION_SEARCH_SORTS.options}
      onSelectOption={() => {}}
    />);
    expect(wrapper.instance().props.id).toBe(2);
  });

  it('can select an option', () => {
    wrapper = shallow(<SelectForm
      id={2}
      label="Some label"
      defaultSort=""
      options={POSITION_SEARCH_SORTS.options}
      onSelectOption={() => {}}
    />);
    wrapper.find('select').simulate('change', { target: { value: 25 } });
    expect(wrapper.instance().state.selection).toBe(25);
  });

  it('can call the selectOption function', () => {
    wrapper = shallow(<SelectForm
      id={1}
      label="Some label"
      defaultSort=""
      options={POSITION_SEARCH_SORTS.options}
      onSelectOption={() => {}}
    />);
    // define the instance
    const instance = wrapper.instance();
    // spy the selectOption function
    const handleSelectSpy = sinon.spy(instance, 'selectOption');
    instance.selectOption({ target: { value: '' } });
    sinon.assert.calledOnce(handleSelectSpy);
  });

  it('matches snapshot', () => {
    wrapper = shallow(<SelectForm
      id={1}
      label="Some label"
      defaultSort=""
      options={POSITION_SEARCH_SORTS.options}
      onSelectOption={() => {}}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
