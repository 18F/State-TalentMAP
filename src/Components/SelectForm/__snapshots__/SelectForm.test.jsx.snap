// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectForm matches snapshot 1`] = `
<div
  className="usa-form"
>
  <label
    htmlFor={1}
  >
    Some label
  </label>
  <select
    id={1}
    name={1}
    onChange={[Function]}
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      Select a sort option
    </option>
    <option
      value="grade"
    >
      Grade (Low to High)
    </option>
    <option
      value="position_number"
    >
      Position Number (0 - Z)
    </option>
    <option
      value="create_date"
    >
      Position Posted Date (New - Old)
    </option>
    <option
      value="post__has_service_needs_differential"
    >
      Service Need Differential (T - F)
    </option>
    <option
      value="bureau__long_description"
    >
      Bureau (A - Z)
    </option>
  </select>
</div>
`;
